Source/ISHRemote/Trisoft.ISHRemote/Cmdlets/BackgroundTask/AddIshBackgroundTask.cs

```diff
@@ -0,0 +1,214 @@
+/*
+* Copyright (c) 2014 All Rights Reserved by the SDL Group.
+* 
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+* 
+*     http://www.apache.org/licenses/LICENSE-2.0
+* 
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using Trisoft.ISHRemote.Objects;
+using Trisoft.ISHRemote.Objects.Public;
+using Trisoft.ISHRemote.Exceptions;
+using Trisoft.ISHRemote.HelperClasses;
+using System.Linq;
+using Trisoft.ISHRemote.BackgroundTask25ServiceReference;
+using Trisoft.ISHRemote.EventMonitor25ServiceReference;
+
+namespace Trisoft.ISHRemote.Cmdlets.BackgroundTask
+{
+    /// <summary>
+    /// <para type="synopsis">Add BackgroundTask.</para>
+    /// <para type="description">Adds BackgroundTask in use cases: </para>
+    /// <para type="description">ParameterGroup uses BackgroundTask25 API to create BackgroundTask.</para>
+    /// <para type="description">IshObjectsGroup uses content object(s) passed via pipeline or as a parameter to trigger BackgroundTask event using DocumentObj25.RaiseEventByIshLngRefs.</para>
+    /// </summary>
+    /// <example>
+    /// <code>
+    /// New-IshSession -WsBaseUrl "https://example.com/ISHWS/" -PSCredential "Admin"
+    /// $ishBackgroundTask = Get-IshFolderContent -FolderPath "General\MyFolder" -VersionFilter Latest -Recurse | Add-IshBackgroundTask -EventType "SMARTTAG"
+    /// </code>
+    /// <para>Add BackgroundTask with event type"SMARTTAG" for the objects located under the "General\MyFolder" path</para> 
+    /// </example>
+    /// <example>
+    /// <code>
+    /// New-IshSession -WsBaseUrl "https://example.com/ISHWS/" -PSCredential "Admin"
+    /// $ishBackgroundTask  = Add-IshBackgroundTask -EventType "PUSHTRANSLATIONS" -EventDescription "Custom event description" -RawInputData $rawData
+    /// </code>
+    /// <para>Add background task with the event type "PUSHTRANSLATIONS" and provided event description and input data</para> 
+    /// </example>
+
+    [Cmdlet(VerbsCommon.Add, "IshBackgroundTask", SupportsShouldProcess = false)]
+    [OutputType(typeof(IshBackgroundTask))]
+    public sealed class AddIshBackgroundTask : BackgroundTaskCmdlet
+    {
+        /// <summary>
+        /// <para type="description">The IshSession variable holds the authentication and contract information. This object can be initialized using the New-IshSession cmdlet.</para>
+        /// </summary>
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = false, ParameterSetName = "ParameterGroup")]
```

Comment:
Missing optional parameter StartAfter  and HashId

Reply:
@ddemeyer I was using the JIRA story and there we still added it...so I thought we missed something from the design. I totally agree not to do the hashId (too technical probably will not be used), but for automating the creation of background tasks I think the StartAfter is very useful.

Reply:
It looks now like a "to be discussed on a call" topic.

Reply:
So we will add -StartAfter (copy -ModifiedSince date time handling from Get-IshEvent), do remember that a different API call needs to be used

Reply:
Done